src/connection.c

```diff
@@ -22,49 +22,87 @@
 #include "config.h"
 #include "power/vita.h"
 #include "input/vita.h"
+#include "video/vita.h"
+#include "audio/vita.h"
 
 #include <stdio.h>
 #include <stdbool.h>
 
 static int connection_status = LI_READY;
 
-void connection_connection_started()
-{
-  connection_status = LI_CONNECTED;
+int connection_failed_stage = 0;
+long connection_failed_stage_code = 0;
+
+void stop_output() {
+  vitainput_stop();
+  vitapower_stop();
+  vitavideo_stop();
+  vitaaudio_stop();
+}
+
+void start_output() {
   vitainput_start();
   vitapower_start();
+  vitavideo_start();
+  vitaaudio_start();
 }
 
-void connection_connection_terminated()
-{
-  vitainput_stop();
-  vitapower_stop();
-  connection_status = LI_DISCONNECTED;
-  quit();
+void connection_connection_started() {
+  connection_status = LI_CONNECTED;
+  start_output();
+}
+
+void connection_connection_terminated() {
+  stop_output();
+  connection_status = LI_READY;
 }
 
-void connection_display_message(char *msg)
-{
+void connection_display_message(char *msg) {
   printf("%s\n", msg);
 }
 
-void connection_display_transient_message(char *msg)
-{
+void connection_display_transient_message(char *msg) {
   printf("%s\n", msg);
 }
 
 void connection_reset() {
+  connection_connection_terminated();
   connection_status = LI_READY;
 }
 
+void connection_minimize() {
+  stop_output();
+  connection_status = LI_MINIMIZED;
+}
+
+void connection_resume() {
+  start_output();
+  connection_status = LI_CONNECTED;
+}
+
+void connection_terminate() {
+  connection_connection_terminated();
+  LiStopConnection();
+  connection_status = LI_READY;
+}
+
+void stage_failed(int stage, long code) {
```

Comment:
i think, it should allow naming convention in this file. maybe `connection_stage_failed`?